packages/examples/cvat/exchange-oracle/tests/api/test_exchange_api.py

```diff
@@ -0,0 +1,283 @@
+    # With wallet address
+    response = client.get(
+        "/tasks", headers={"signature": "sample"}, params={"wallet_address": user_address}
+    )
+
+    assert response.status_code == 200
+    assert isinstance(response.json(), list)
+    assert len(response.json()) == 0
+
+    # Without wallet address
+    response = client.get(
+        "/tasks",
+        headers={"signature": "sample"},
+    )
+
+    assert response.status_code == 200
+    assert isinstance(response.json(), list)
+    assert len(response.json()) == 0
+
+    with (SessionLocal.begin() as session,):
+        _, _, cvat_job_1 = create_project_task_and_job(
+            session, "0x86e83d346041E8806e352681f3F14549C0d2BC67", 1
+        )
+        create_project_task_and_job(session, "0x86e83d346041E8806e352681f3F14549C0d2BC68", 2)
+        create_project_task_and_job(session, "0x86e83d346041E8806e352681f3F14549C0d2BC69", 3)
+
+        user = User(
+            wallet_address=user_address,
+            cvat_email="test@hmt.ai",
+            cvat_id=1,
+        )
+        session.add(user)
+
+        assignment = Assignment(
+            id=str(uuid.uuid4()),
+            user_wallet_address=user_address,
+            cvat_job_id=cvat_job_1.cvat_id,
+            expires_at=datetime.now() + timedelta(days=1),
+        )
+        session.add(assignment)
+
+        session.commit()
+
+        with (
+            open("tests/utils/manifest.json") as data,
+            patch("src.services.exchange.get_escrow_manifest") as mock_get_manifest,
+        ):
+            manifest = json.load(data)
+            mock_get_manifest.return_value = manifest
+
+            # With wallet address
+            response = client.get(
+                "/tasks", headers={"signature": "sample"}, params={"wallet_address": user_address}
+            )
+
+            assert response.status_code == 200
+            assert isinstance(response.json(), list)
+            assert len(response.json()) == 1
+            for task in response.json():
+                assert task["assignment"]
+                assert set(task.keys()) == {
+                    "id",
+                    "escrow_address",
+                    "title",
+                    "description",
+                    "platform",
+                    "job_bounty",
+                    "job_size",
+                    "job_time_limit",
+                    "job_type",
+                    "assignment",
+                    "status",
+                }
+
+            # Without wallet address
+            response = client.get(
+                "/tasks",
+                headers={"signature": "sample"},
+            )
+
+            assert response.status_code == 200
+            assert isinstance(response.json(), list)
+            assert len(response.json()) == 2
+            for task in response.json():
+                assert not task["assignment"]
+                assert set(task.keys()) == {
```

Comment:
In CVAT we use the DeepDiff library to compare such data in tests. Consider using it here too.

Reply:
In that data we were not comparing the data itself, we just wanted to make sure that the data returned were instances of the TaskReturned class. I tried that with the isInstance method but didn't get it. So I'm just checking that the object contains some specific keys.